bin/dune_init.ml

```diff
@@ -519,7 +519,43 @@ module Component = struct
             [ File.make_text (Path.parent_exn opam_file) (Path.basename opam_file) "" ]
           | Esy -> [ File.make_text dir "package.json" "" ]
         in
-        { dir; files = dune_project_file dir opts :: package_files }
+        let default_files =
+          let fmt_version =
+            try
+              let ic = Unix.open_process_in "ocamlformat --version" in
+              let version = input_line ic in
+              let _ = Unix.close_process_in ic in
+              (let open Pp.O in
+              Console.print_user_message
+                (User_message.make
+                   [ Pp.tag User_message.Style.Hint (Pp.verbatim "Format")
+                     ++ Pp.textf ": using ocamlversion number %s" version
```

Comment:
Is this meant to be

```suggestion
                     ++ Pp.textf ": using ocamlformat version number %s" version
```

I'd ask us to be careful about adding output, and push back a bit to ask why we need to output anything at all here. The reasons to be careful here is that this kind of thing can quickly adds up to a wall of spam. I don't know whether it is an explicit part of the design philosophy, but dune tends to mostly follow the [Unix Rule of Silence](https://www.linfo.org/rule_of_silence.html), and I suggest we do that here as well.

This output could be moved into the debugging logs, but I'm not evens sure why we'd need it there. WDYT?

Reply:
One other thought here: currently `Dune_init` is designed to push all the I/O operations to the edge, and only have pure computations deriving all of the components. The main benefit is that it makes it easy to reason about compositions of different component generation logic. The addition of this interaction with the `ocamlformat` executable in the middle of deriving the project violates this design approach. Perhaps we can find a way to move this kind of action into a configuration phase, rather then during component derivation? I'd even suggest moving this kind of I/O into https://github.com/ocaml/dune/tree/main/bin/init.ml , as this seems analogous to our creation of the new project directory and initilization of the build context to my mind. But I wonder if core maintainers think that is the right place?